backend/WebUI/api_webui.go

```diff
@@ -1952,3 +1954,205 @@ func OptionsSubscribers(c *gin.Context) {
+	profile, err := mongoapi.RestfulAPIGetOne(profileDataColl, bson.M{"profileName": profileName})
+	if err != nil {
+		logger.ProcLog.Errorf("GetProfile err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	var pf Profile
+	err = json.Unmarshal(mapToByte(profile), &pf)
+	if err != nil {
+		logger.ProcLog.Errorf("JSON Unmarshal err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	c.JSON(http.StatusOK, pf)
+}
+
+// Post profile
+func PostProfile(c *gin.Context) {
+	setCorsHeader(c)
+	logger.ProcLog.Infoln("Post One Profile Data")
+
+	tokenStr := c.GetHeader("Token")
+	_, err := ParseJWT(tokenStr)
+	if err != nil {
+		logger.ProcLog.Errorln(err.Error())
+		c.JSON(http.StatusBadRequest, gin.H{
+			"cause": "Illegal Token",
+		})
+		return
+	}
+
+	var profile Profile
+	if err = c.ShouldBindJSON(&profile); err != nil {
+		logger.ProcLog.Errorf("PostProfile err: %+v", err)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"cause": "JSON format incorrect",
+		})
+		return
+	}
+
+	tenantData, err := mongoapi.RestfulAPIGetOne(tenantDataColl, bson.M{"tenantName": "admin"})
+	if err != nil {
+		logger.ProcLog.Errorf("GetProfile err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	profile.TenantId = tenantData["tenantId"].(string)
+
+	pf, err := mongoapi.RestfulAPIGetOne(profileListColl, bson.M{"profileName": profile.ProfileName})
+	if err != nil {
+		logger.ProcLog.Errorf("GetProfile err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	if len(pf) != 0 {
+		c.JSON(http.StatusConflict, gin.H{
+			"cause": "Profile already exists",
+		})
+		return
+	}
+
+	logger.ProcLog.Infof("PostProfile: %+v", profile.ProfileName)
+	dbProfileOperation(profile.ProfileName, "post", &profile)
+	c.JSON(http.StatusCreated, gin.H{})
+}
+
+// Put profile by profileName
+func PutProfile(c *gin.Context) {
+	setCorsHeader(c)
+	logger.ProcLog.Infoln("Put One Profile Data")
+
+	profileName := c.Param("profileName")
+
+	var profile Profile
+	if err := c.ShouldBindJSON(&profile); err != nil {
+		logger.ProcLog.Errorf("PutProfile err: %+v", err)
+		c.JSON(http.StatusBadRequest, gin.H{})
+		return
+	}
+
+	pf, err := mongoapi.RestfulAPIGetOne(profileListColl, bson.M{"profileName": profile.ProfileName})
+	if err != nil {
+		logger.ProcLog.Errorf("PutProfile err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	if len(pf) == 0 {
+		c.JSON(http.StatusNotFound, gin.H{
+			"cause": "Profile does not exist",
+		})
+		return
+	}
+
+	tenantData, err := mongoapi.RestfulAPIGetOne(tenantDataColl, bson.M{"tenantName": "admin"})
+	if err != nil {
+		logger.ProcLog.Errorf("GetProfile err: %+v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{})
+		return
+	}
+	profile.TenantId = tenantData["tenantId"].(string)
+
+	logger.ProcLog.Infof("PutProfile: %+v", profile.ProfileName)
+	dbProfileOperation(profileName, "put", &profile)
+	c.JSON(http.StatusNoContent, gin.H{})
```

Comment:
I think use the `http.StatuOK ` is better than use the `http.StatueNoContext` in this case.
Besides, You should return the updated data instead of return `gin.H{}`.